From 443e26852129e181a6cead6364a58ae64c1bcb63 Mon Sep 17 00:00:00 2001
From: Wang Hui <Hui.Wang@windriver.com>
Date: Wed, 27 May 2009 09:58:50 +0800
Subject: [PATCH 69/77] imx27_mach/one_wire: add one_wire registration in platform init process

Add CONFIG_W1_MASTER_MXC_MODULE as a condition to register the one_wire
master device to the platform bus.
It seems strange and unreasonable that the static kernel depends on a
MODULE's choice, but this situation is common among platform init
process. Because the MODULE(the device driver) only implement
platform_driver, if the driver want to work, a corresponding
platform_device must be registered. This registration action always
happen in the platform init process rather than in the MODULE's self.

Signed-off-by: Wang Hui <Hui.Wang@windriver.com>
---
 arch/arm/mach-mx2/mx27ads.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx2/mx27ads.c b/arch/arm/mach-mx2/mx27ads.c
index 024d580..a56e8a4 100644
--- a/arch/arm/mach-mx2/mx27ads.c
+++ b/arch/arm/mach-mx2/mx27ads.c
@@ -1377,7 +1377,7 @@ struct mxc_usb2_platform_data ehci2_pdata = {
 };
 #endif
 
-#ifdef CONFIG_W1_MASTER_MXC
+#if defined(CONFIG_W1_MASTER_MXC) || defined(CONFIG_W1_MASTER_MXC_MODULE)
 static struct platform_device mx27ads_w1_device = {
 	.name = "mxc_w1",
 	.id = 0
@@ -1559,7 +1559,7 @@ static struct platform_device *platform_devices[] __initdata = {
 #ifdef CONFIG_KEYBOARD_MXC
 	&mx27ads_keypad_device,
 #endif /* KEYBOARD_MXC */
-#ifdef CONFIG_W1_MASTER_MXC
+#if defined(CONFIG_W1_MASTER_MXC) || defined(CONFIG_W1_MASTER_MXC_MODULE)
 	&mx27ads_w1_device,
 #endif /* W1_MASTER_MXC */
 };
-- 
1.6.3.1

